### PR TITLE
Don't recreate ResizeObserver on callback reference change

### PR DESCRIPTION
**Pull request by leroykorterink**
See title

### CODE REVIEW

File: src/hooks/useResizeObserver/useResizeObserver.ts [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
 /**
- * This hook allows you to add a ResizeObserver for an element and remove it
+ * This hook allows you to add a ResizeObserver to an element and remove it
  * when the component unmounts.
  *
  * @param ref - The ref to observe
  * @param callback - The callback to fire when the element resizes
  */
-export function useResizeObserver(ref: RefObject<Element>, callback: ResizeObserverCallback): void {
+export function useResizeObserver(
```
Comment (ThaNarie) [inline src/hooks/useResizeObserver/useResizeObserver.ts:12]: Do we want a separate ticket to support `unobserve`, e.g. having a boolean arg that defaults to `true`, but can be set to `false` to unobserve ?

Comment (leroykorterink) [reply src/hooks/useResizeObserver/useResizeObserver.ts:12]: I think this behaviour is very unlikely and I don't know if I like a 4th parameter for the hook function signature.

```ts
function useResizeObserver(
  target: Unreffable<Element>,
  callback: ResizeObserverCallback,
  options?: ResizeObserverOptions,
  enabled = true,
): void
```

Implementation will be similar to this:

```ts
export function useResizeObserver(
  target: Unreffable<Element>,
  callback: ResizeObserverCallback,
  options?: ResizeObserverOptions,
  enabled = true,
): void {
  const memoizedOptions = useMemo(
    () => options,
    [JSON.stringify(options)],
  );

  const callbackRef = useRefValue(callback);
  const resizeObserver = useClientSideValue(
    () =>
      new ResizeObserver((entries, observer) => {
        callbackRef.current?.(entries, observer);
      }),
  );

  useEffect(() => {
    const element = unref(target);

    if (element === null) {
      return;
    }

    if (enabled) {
      resizeObserver?.observe(element, memoizedOptions);
    } else {
      resizeObserver?.unobserve(element);
    }

    return () => {
      resizeObserver?.unobserve(element);
    };
  }, [enabled, memoizedOptions, resizeObserver, target]);
}
```

File: src/hooks/useResizeObserver/useResizeObserver.ts [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
     }
 
-    resizeObserverInstance.observe(ref.current);
+    const resizeObserverInstance = new ResizeObserver(
+      (entries: Array<ResizeObserverEntry>, observer: ResizeObserver) => {
+        callbackRef.current?.(entries, observer);
+      },
+    );
+
+    resizeObserverInstance.observe(element);
```
[the remark this answers is not in the document]
Comment (leroykorterink) [reply src/hooks/useResizeObserver/useResizeObserver.ts:31]: Created an issue https://github.com/mediamonks/react-hooks/issues/162

**Comment by leroykorterink** [comment]
Feature is enhanced in #274

**Pull request closed without merging**